### PR TITLE
Add Gray-Scott demo

### LINKED ISSUE #118: Add Gray-Scott demo
It would be useful to have a demo with two solves that are different.

### PR DESCRIPTION
**Pull request by jwallwork23**
Closes #118.

Add a demo for the Gray-Scott system, solved as two separate equations. Shows how to account for the 'split' case where the equations differ.

### CODE REVIEW

**Comment by ddundo** [comment]
Thanks for this @jwallwork23! Could you please expand on what would be the best way to do goal-based error estimation here? In particular, I'm looking at [evaluating the error indicator here](https://github.com/pyroteus/pyroteus/blob/6368db9d16949b89943625818a14abbfeb8fdc69/pyroteus/go_mesh_seq.py#L189) which takes a single form from `get_form`.

**Comment by jwallwork23** [comment]
Good point. In my recent commits I change the way `get_form` works. It now needs to return a dictionary, whose keys are the (strongly-coupled) fields, i.e., the equations. Let me know what you think.

**Comment by jwallwork23** [comment]
Ah yes, good spot @ddundo. If you could fix the [errors ](https://github.com/pyroteus/pyroteus/actions/runs/4553879300/jobs/8030998218?pr=119) you introduced in the CI then that'd be much appreciated, thanks.

File: pyroteus/go_mesh_seq.py [hunk 1]
```diff
@@ -137,13 +138,26 @@ def transfer(source, target):
...
+            }
+        )
+
         FWD, ADJ = "forward", "adjoint"
         FWD_OLD = "forward" if self.steady else "forward_old"
         ADJ_NEXT = "adjoint" if self.steady else "adjoint_next"
         for i, mesh in enumerate(self):
-            P0 = FunctionSpace(self[i], "DG", 0)
-            indicator = []
+            P0 = P0_spaces[i]
```
Comment (jwallwork23) [inline pyroteus/go_mesh_seq.py:160]: @ddundo You don't need to define this `P0` temporary if you only use it once.

**Review by ddundo — APPROVED** [review]

**Comment by jwallwork23** [comment]
Thanks for your efforts @ddundo. Merged and updated the website!

**Pull request merged**